src/components/legend/color.tsx

```diff
@@ -34,32 +33,34 @@ interface ColorProps {
  * @internal
  */
 export const Color = memo(
-  forwardRef<HTMLDivElement, ColorProps>(({ color, isSeriesHidden = false, hasColorPicker, onClick }, ref) => {
+  forwardRef<HTMLButtonElement, ColorProps>(({ color, isSeriesHidden = false, hasColorPicker, onClick }, ref) => {
     if (isSeriesHidden) {
       return (
-        <button type="button" className="echLegendItem__color" aria-label="series hidden" title="series hidden">
+        <div className="echLegendItem__color" title="series hidden">
           {/* changing the default viewBox for the eyeClosed icon to keep the same dimensions */}
-          <Icon type="eyeClosed" viewBox="-3 -3 22 22" />
-        </button>
+          <Icon type="eyeClosed" viewBox="-3 -3 22 22" aria-label="series color is hidden" />
+        </div>
       );
     }
 
-    const colorClasses = classNames('echLegendItem__color', {
-      'echLegendItem__color--changable': hasColorPicker,
-    });
+    if (hasColorPicker) {
+      return (
+        <button
+          type="button"
+          onClick={onClick}
+          className="echLegendItem__color--changable"
+          title="change series color"
+          ref={ref}
+        >
+          <Icon type="dot" color={color} aria-label="change series color" />
```

Comment:
Are names for these colors defined anywhere? Would be good to put that into the label. Worst case, even putting the hex code would be better than nothing... 

```suggestion
          <Icon type="dot" color={color} aria-label="change series color, currently ${colorName}" />
```

Reply:
We don't have a naming mapping for the colors as these are actually passed to the charts by the consumer. A conversion from hex to actual name is out of the scope of the chart lib at the moment. Something like that can be built and shared across all kibana, a approximation mapping like the one provided by https://chir.ag/projects/name-that-color/#6195ED can be an interesting solution and can go into the same direction of the i18n

Reply:
Then hex code is fine for now

Reply:
Thank you for catching this - edited in 07eadda